rostime/src/time.cpp

```diff
@@ -167,6 +167,29 @@ namespace ros
     nsec = nsec_sum;
 #endif
   }
+
+  void ros_monotonictime(uint32_t& sec, uint32_t& nsec)
+#ifndef WIN32
+    throw(NoHighPerformanceTimersException)
+#endif
+  {
+#ifndef WIN32
+#if HAS_CLOCK_GETTIME
+    timespec start;
+    clock_gettime(CLOCK_MONOTONIC, &start);
+    sec  = start.tv_sec;
+    nsec = start.tv_nsec;
+#else
+    // what to do if clock_gettime is not available???
```

Comment:
Correct. I would actually just leave the case when `clock_gettime` is not available out completely, as I don't know of any way to get a monotonic clock otherwise and this should not be the case on any actually used Linux versions anymore...